internal/tfimport/tfimport.go

```diff
@@ -538,19 +573,52 @@ func Run(rn runner.Runner, importRn runner.Runner, runArgs *RunArgs) error {
 		return fmt.Errorf("expand path %q: %v", inputDir, err)
 	}
 
+	var successesTotal, failuresTotal []string
 	skipped := make(map[string]bool)
+	var ie *importErr
 	for {
-		retry, err := planAndImport(rn, importRn, runArgs, skipped)
+		failuresTotal = nil
+		successes, err := planAndImport(rn, importRn, runArgs, skipped)
+		successesTotal = append(successesTotal, successes...)
+
 		if err != nil {
-			return err
+			if errors.As(err, &ie) {
+				failuresTotal = ie.errMsgs
+				if len(successes) > 0 {
+					// Some resources imported successfully, but others didn't.
+					// Time to retry.
+					log.Println("Some imports succeeded but others did not. Retrying the import, in case dependent values have now been populated.")
+					continue
+				}
+			} else {
+				// Otherwise it's some other kind of error and we should return it.
+				return err
+			}
 		}
 
-		// If all imports succeeded, or none did, stop trying to import things, since we aren't making progress anymore.
-		if !retry {
-			break
-		}
+		// Either no successes, or no errors.
+		// No need to retry.
+		break
+	}
+
+	var skippedS []string
+	for resource := range skipped {
+		skippedS = append(skippedS, resource)
+	}
+
+	buf, err := template.WriteBuffer(summaryTmpl, map[string]interface{}{
+		"total":     len(successesTotal) + len(skippedS) + len(failuresTotal),
+		"successes": successesTotal,
+		"skipped":   skippedS,
```

Comment:
Why not pass skipped as a map to the template?

Reply:
I thought it wouldn't work, but I tried it and it does, thanks!